src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java

```diff
@@ -0,0 +1,306 @@
+            fromField.setText(String.valueOf(payloadGeneratorUI.getPayloadGenerator().getFrom()));
+            toField.setText(String.valueOf(payloadGeneratorUI.getPayloadGenerator().getTo()));
+            stepField.setText(String.valueOf(payloadGeneratorUI.getPayloadGenerator().getStep()));
+        }
+
+        @Override
+        public NumberPayloadGeneratorUI getPayloadGeneratorUI() {
+            if (oldGenerator != null) {
+                return oldGenerator;
+            }
+
+            return  new NumberPayloadGeneratorUI(getPayloadGenerator());
+
+        }
+
+        @Override
+        public void clear() {
+            oldGenerator = null;
+            getPayloadsPreviewTextArea().setText("");
+        }
+
+        @Override
+        public boolean validate() {
+            return true;
+        }
+
+        @Override
+        public String getHelpTarget() {
+            return null;
+        }
+
+        private void setPreviewAndSaveButtonsEnabled(boolean enabled) {
+            getPayloadsPreviewGenerateButton().setEnabled(enabled);
+            getSaveButton().setEnabled(enabled);
+        }
+
+        private JButton getPayloadsPreviewGenerateButton() {
+            if (payloadsPreviewGenerateButton == null) {
+                payloadsPreviewGenerateButton = new JButton(PAYLOADS_PREVIEW_GENERATE_FIELD_LABEL);
+                payloadsPreviewGenerateButton.setEnabled(false);
+
+                payloadsPreviewGenerateButton.addActionListener(new ActionListener() {
+
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        updatePayloadsPreviewTextArea();
+                    }
+                });
+            }
+            return payloadsPreviewGenerateButton;
+        }
+
+        private JTextArea getPayloadsPreviewTextArea() {
+            if (payloadsPreviewTextArea == null) {
+                payloadsPreviewTextArea = new JTextArea(15, 10);
+                payloadsPreviewTextArea.setEditable(false);
+            }
+            return payloadsPreviewTextArea;
+        }
+
+        private void updatePayloadsPreviewTextArea() {
+            NumberPayloadGenerator payloadGenerator = getPayloadGenerator();
+            if (payloadGenerator == null) {
+                return;
+            }
+            StringBuilder contents = new StringBuilder();
+            try {
+                try (ResettableAutoCloseableIterator<DefaultPayload> payloads = payloadGenerator.iterator()) {
+
+                    for (int i = 0; i < MAX_NUMBER_PAYLOADS_PREVIEW && payloads.hasNext(); i++) {
+                        if (contents.length() > 0) {
+                            contents.append('\n');
+                        }
+                        contents.append(payloads.next().getValue());
+                    }
+                }
+                getPayloadsPreviewTextArea().setEnabled(true);
+            } catch (Exception ignore) {
```

Comment:
Can this happen? Looking at the iterator's implementation it does not produce any exception.